Main.py

```diff
@@ -2,431 +2,368 @@
-                msg += "~ %s's Regular Crossword Avg: %d:%s [%s]~\n" % (member.name,minutes,str(seconds).zfill(2),avg_diff)
-            else:
-                msg += "~ %s's Regular Crossword Avg: %d:%s ~\n" % (member.name,minutes,str(seconds).zfill(2))
-        else:
-            if avg_diff:
-                msg += "~ %s's Regular Crossword Avg: %d [%s]~\n" % (member.name,int(new_reg_avg),avg_diff)
-            else:
-                msg += "~ %s's Regular Crossword Avg: %d ~\n" % (member.name,int(new_reg_avg))
-        c.execute("INSERT OR REPLACE INTO Ranking VALUES({0},'{1}',{2},(SELECT SatAvg FROM Ranking WHERE ID={0}))".format(member.id,member.name,new_reg_avg))
-        conn.commit()
-        avg_diff = None
-    if saturdays != 0:
-        new_sat_avg = int(sat_avg/saturdays)
-        try:
-            old_sat_avg = avgslist[0][1] 
-        except:
-            old_sat_avg = None
-        if old_sat_avg != None:
-            if day == "Sat":
-                avg_diff = str(new_sat_avg - old_sat_avg)
-                if avg_diff[0] != "-": avg_diff = "+" + avg_diff
-        if new_sat_avg > 59:
-            minutes = math.floor(new_sat_avg/60)
-            seconds = int(new_sat_avg % 60)
-            if avg_diff:
-                msg += "~ %s's Saturday Crossword Avg: %d:%s [%s]~" % (member.name,minutes,str(seconds).zfill(2),avg_diff)
-            else:
-                msg += "~ %s's Saturday Crossword Avg: %d:%s ~" % (member.name,minutes,str(seconds).zfill(2))
-        else:
-            if avg_diff:
-                msg += "~ %s's Saturday Crossword Avg: %d [%s]~" % (member.name,int(new_sat_avg),avg_diff)
-            else:
-                msg += "~ %s's Saturday Crossword Avg: %d ~" % (member.name,int(new_sat_avg))
-        c.execute("INSERT OR REPLACE INTO Ranking VALUES({0},'{1}',(SELECT RegAvg FROM Ranking WHERE ID={0}),{2})".format(member.id,member.name,new_sat_avg))
+        day = _get_day(datestamp)
+
+        c.execute("INSERT OR REPLACE INTO Scores VALUES (?, ?, ?, ?)",
+                  (member.id, member.name, _as_ymd(datestamp), time))
         conn.commit()
+        await ctx.send("```css\nScore added.\n```")
+
+        # calculate new avg
+        times_list = c.execute('SELECT Score, Date FROM Scores WHERE ID = ?', (member.id,)).fetchall()
+        reg_vals = []
+        sat_vals = []
+        for score, score_date in times_list:
+            (sat_vals if _get_day_from_ymd(score_date) == "Sat" else reg_vals).append(int(score))
+
+        avgs_list = c.execute('SELECT RegAvg, SatAvg FROM Ranking WHERE ID = ?', (member.id,)).fetchall()
+
+        def _format_avg_delta(ad):
+            return f"[{ad}]" if ad else ""
+
+        msg = "```"
+
+        if reg_vals:
+            new_reg_avg = int(statistics.mean(reg_vals))
+            old_reg_avg = avgs_list[0][0] if avgs_list else None
+            avg_diff = f"{str(new_reg_avg - old_reg_avg):+d}" if old_reg_avg is not None and day != "Sat" else None
```

Comment:
Unknown format code 'd' for object of type 'str'

Reply:
19e11bd1e01522040608edc461f68e8d7edf8cc6